File: LeetCode/java/2287_Rearrange_Characters_to_Make_Target_String.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int rearrangeCharacters(String s, String target) {
        int answer = 100;
        Map<Character, Integer> map = new HashMap<>();
        Map<Character, Integer> map2 = new HashMap<>();

        for (int i = 0; i < s.length(); i++) {
            map.put(s.charAt(i), map.getOrDefault(s.charAt(i), 0) + 1);
        }

        for (int i = 0; i < target.length(); i++) {
            map2.put(target.charAt(i), map2.getOrDefault(target.charAt(i), 0) + 1);
        }

        for (char key : map2.keySet()) {
            answer = Math.min(answer, map.getOrDefault(key, 0) / map2.get(key));
        }

        return answer;
    }
}

File: LeetCode/cpp/2287_Rearrange_Characters_to_Make_Target_String.cpp
class Solution {
public:
	int rearrangeCharacters(string s, string target) {
		int answer = 100;
		unordered_map<char, int> um, um2;

		for (int i = 0; i < s.length(); i++) {
			um[s[i]]++;
		}

		for (int i = 0; i < target.length(); i++) {
			um2[target[i]]++;
		}

		for (auto m : um2) {
			answer = min(answer, um[m.first] / m.second);
		}

		return answer;
	}
};